pkg/azuredx/azureauth/auth_test.go

```diff
@@ -86,6 +86,32 @@ func TestOnBehalfOf(t *testing.T) {
 	}
 }
 
+func TestOnBehalfOfDisabled(t *testing.T) {
+	// compose request
+	var req backend.QueryDataRequest
+	req.PluginContext.User = &backend.User{Login: "alice"}
+	req.Headers = make(map[string]string)
+	req.Headers["X-ID-Token"] = "ID-TOKEN"
+
+	// setup & test
+	fakeTokenProvider := &FakeTokenProvider{}
+
+	c := &ServiceCredentialsImpl{
+		tokenProvider:  fakeTokenProvider,
+	}
+	c.OnBehalfOf = false
+
+	auth, err := c.QueryDataAuthorization(context.Background(), &req)
+
+	if err != nil {
+		t.Errorf("got error %q", err)
+	}
+
+	if !fakeTokenProvider.TokenRequested {
+		t.Errorf("got %q, expected 'TokenRequested", auth)
+	}
```

Comment:
same: `assert.Equal(t, fakeTokenProvider.TokenRequested, true)`

but for clarity with the previous set of tests, shouldn't you use a FakeAADClient and check that `TokenRequested` is false? (Therefore, `AcquireTokenOnBehalfOf` has not been called). 

What I would do is to extend the table test above, make `OnBehalfOf` a test input and check that the token has not been requested. Something like:

```
		4: {
			RequestUser:          &backend.User{Login: "alice"},
			RequestIDTokenHeader: "ID-TOKEN",
			OnBehalfOfEnabled: false,
			ShouldRequestToken:   false,
			ExpectedError:        "",
		},
```
...
```
	c.OnBehalfOf = g.OnBehalfOfEnabled
```